lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2025,8 +2025,18 @@ func (d *zfs) migrateVolumeOptimized(vol Volume, conn io.ReadWriteCloser, volSrc
 					finalParent = volSrcArgs.Data.(map[ContentType]string)[vol.ContentType()]
 				}
 
-				defer func() { _, _ = shared.RunCommand("zfs", "destroy", finalParent) }()
-				defer func() { _, _ = shared.RunCommand("zfs", "destroy", srcSnapshot) }()
+				defer func() {
+					_, err := shared.RunCommand("zfs", "destroy", finalParent)
+					if err != nil {
+						d.logger.Warn("Failed deleting final sync temporary snapshot parent", logger.Ctx{"snapshot": finalParent, "err": err})
+					}
+				}()
+				defer func() {
+					_, err := shared.RunCommand("zfs", "destroy", srcSnapshot)
+					if err != nil {
+						d.logger.Warn("Failed deleting final sync temporary snapshot", logger.Ctx{"snapshot": srcSnapshot, "err": err})
+					}
+				}()
```

Comment:
This entire branch can actually be removed, as `volSrcArgs.FinalSync` will always be `false.`

Reply:
Yep doing now infact :)

Reply:
@monstermunchkin done.